pkg/build/controller/build/build_controller.go

```diff
@@ -476,6 +476,12 @@ func shouldIgnore(build *buildv1.Build) bool {
 		return true
 	}
 
+	// builds that are in the process of being deleted should be ignored, otherwise
+	// they might attempt to recreate a pod that was just GCed as part of the build
+	// deletion propagation.
+	if build.DeletionTimestamp != nil {
+		return true
+	}
```

Comment:
@bparees unit test? Feels like we should add one since there are a few cases in this function...

Reply:
can you get someone to do it as a followup? i don't know when i'm going to have time to come back to this.

Reply:
:+1: - will create a GH issue